cpp/src/arrow/array/diff.cc

```diff
@@ -664,7 +663,23 @@ class MakeFormatterImpl {
     return Status::OK();
   }
 
-  Status Visit(const DataType& t) {
+  Status Visit(const NullType& t) {
```

Comment:
Diff between null arrays is supported. I think there's a unit test too

Reply:
I think it's supported, just not the Formatter.